src/components/Auth/ResetPassword/ForgotPassForm.tsx

```diff
@@ -0,0 +1,111 @@
+'use client'
+
+import Loader from '@/components/UI/Loader/Loader'
+import Button from '@/components/UI/Buttons/Button/Button'
+import FormInput from '@/components/UI/FormInput/FormInput'
+import { useForm } from 'react-hook-form'
+import { useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { IForgotValues } from '@/types/ChangePassword'
+import { apiForgotPassword } from '@/services/userService'
+import { useErrorHandler } from '@/services/apiError/apiError'
+import { forgotPassSchema } from '@/validation/forgotPassSchema'
+import { useLocalSessionStore } from '@/store/useLocalSessionStore'
+
+
+export default function ForgotPassForm() {
+  const [loading, setLoading] = useState(false)
+  const { errorMessage, handleError } = useErrorHandler()
+  const router = useRouter()
+  const setStoredEmailSent = useLocalSessionStore((state) => state.setStoredEmailSent)
+  const storedEmailSent = useLocalSessionStore((state) => state.storedEmailSent)
+
+
+  const { handleSubmit, register, getValues, reset,
+    formState: { errors },
+  } = useForm<IForgotValues>({
+    resolver: yupResolver(forgotPassSchema),
+    defaultValues: {
+      email: ''
+    },
+  })
+
+
+
+  const onSubmit = async () => {
+    const { email } = getValues() as { email: string }
+
+    try {
+      setLoading(true)
+      await apiForgotPassword({ email })
+      setStoredEmailSent(true)
+      localStorage.setItem('emailForReset', email)
```

Comment:
I think email should be saved to a shared local session for more structured code